packages/ui5-config/src/ui5config.ts

```diff
@@ -334,19 +386,121 @@ export class UI5Config {
+     * @memberof UI5Config
+     */
+    public addServiceToMockserverMiddleware(
+        serviceName: string,
+        servicePath: string,
+        appRoot?: string,
+        annotationsConfig: MockserverConfig['annotations'] = []
+    ): this {
+        const middlewareListYaml = this.document.getSequence({ path: 'server.customMiddleware' });
+        // Throw error if middleware is not found
+        const mockserverMiddlewareYaml = this.document.findItem(
+            middlewareListYaml,
+            (item: any) => item.name === 'sap-fe-mockserver'
+        ) as YAMLMap;
+        if (!mockserverMiddlewareYaml) {
+            throw new Error('Could not find sap-fe-mockserver');
+        } else {
+            // Else append new data to current middleware config and then run middleware update
+            const serviceRoot = `${appRoot ?? './webapp'}/localService/${serviceName}`;
+
+            const mockserverMiddleware = this.findCustomMiddleware('sap-fe-mockserver') as CustomMiddleware;
+            const mockserverMiddlewareConfig = mockserverMiddleware?.configuration as MockserverConfig;
+            if (mockserverMiddlewareConfig.services) {
+                const newServiceData = {
+                    urlPath: servicePath,
+                    metadataPath: `${serviceRoot}/metadata.xml`,
+                    mockdataPath: `${serviceRoot}/data`,
+                    generateMockData: true
+                };
+                const serviceIndex = mockserverMiddlewareConfig.services.findIndex(
+                    (existingService) => existingService.urlPath === servicePath || existingService.urlPath === ''
+                );
+                if (serviceIndex > -1) {
+                    mockserverMiddlewareConfig.services[serviceIndex] = newServiceData;
+                } else {
+                    mockserverMiddlewareConfig.services = [...mockserverMiddlewareConfig.services, newServiceData];
+                }
+            }
+            if (mockserverMiddlewareConfig.annotations) {
+                const existingAnnotations = mockserverMiddlewareConfig.annotations;
+                annotationsConfig.forEach((annotationConfig) => {
+                    if (
+                        !existingAnnotations.find(
+                            (existingAnnotation) => existingAnnotation.urlPath === annotationConfig.urlPath
+                        )
+                    ) {
+                        existingAnnotations.push(annotationConfig);
+                    }
+                });
+            }
+            this.updateCustomMiddleware(mockserverMiddleware);
+        }
+        return this;
+    }
+
+    /**
+     * Removes a service from the mockserver middleware.
+     *
+     * @param servicePath path of the service that is to be deleted
+     * @returns {UI5Config} the UI5Config instance
+     * @memberof UI5Config
+     */
+    public removeServiceFromMockServerMiddleware(servicePath: string): this {
```

Comment:
What happens to the annotations? Shouldn't you remove them as well if you remove a service config?

Reply:
Right.
Can we do it this way:

```
            if (mockserverMiddlewareConfig.annotations) {
                const mockserverMiddlewareAnnotations = mockserverMiddlewareConfig.annotations;
                const configuration = this.document.getMap({
                    start: mockserverMiddlewareYaml,
                    path: 'configuration'
                });
                serviceAnnotations.forEach((annotationName: string) => {
                    const annotationIndex = mockserverMiddlewareAnnotations.findIndex((existingAnnotation) => {
                        return existingAnnotation.localPath?.endsWith(`${annotationName}.xml`);
                    });
                    const annotations = this.document.getSequence({ start: configuration, path: 'annotations' });
                    annotations.delete(annotationIndex);
                });
            }
```

Where `serviceAnnotations: string[]` is parameter with annotation names associated with the service.

Reply:
Right.
I have 2 solutions for this:

1. We collect annotations URI while they are getting removed from manifest.
`const serviceAnnotationsURI = deleteServiceFromManifest(basePath, service.name, fs).deletedAnnotationsURI;`

2. We collect annotations URI from service.annotations
```
        const annotations = service.annotations as EdmxAnnotationsInfo | EdmxAnnotationsInfo[];
        const serviceAnnotationsURI: string[] = [];
        if (Array.isArray(annotations)) {
            annotations.forEach((annotation: EdmxAnnotationsInfo) => {
                serviceAnnotationsURI.push(`localService/${annotation.technicalName}.xml`);
            });
        } else {
            serviceAnnotationsURI.push(`localService/${annotations.technicalName}.xml`);
        }
```

Then we pass collected annotations URI to `removeServiceFromMockServerMiddleware` in `ui5-config` which loops through mockserver middleware annotations and deletes ones that are matching with those we collected.